### DISCUSSION TITLE
Decoration line covers Y-axis label

### DISCUSSION THREAD

**Discussion by my-waiwai** [patrykandpatrick/vico#982] [Questions]
### Question

I've been using vico to display stock market data  and I'm trying to play with using decorations to show the previous close and open values (instead of a hacked-together series). It looks to me like the decoration lines have a higher elevation than the Y-axis labels. I have my axis labels inside with a background color that matches the app's main background color (shown in the screenshot). This shows nicely above the line chart but it looks to me that the decoration line cuts through the Y-axis labels. Is there a way to make it so the Y-axis elevation is higher than the decoration lines so that the axis label background will cover part of the line and the text will display cleanly? Just for kicks I changed the thickness to `5.dp` to see what it looked like. Screenshot of that is included.

<img width="284" alt="Screenshot_2" src="https://github.com/user-attachments/assets/9863f9f9-694a-4db0-b580-5e334bede797" />
<img width="284" alt="Screenshot_3" src="https://github.com/user-attachments/assets/770eaaa5-d696-4977-af7e-0b67d2d40637" />

### Vico version(s)

2.0.0-beta.3

### UI framework(s)

Jetpack Compose

**Comment by Gowsky** [comment]
Hello, @my-waiwai. `CartesianChart` can draw `Decoration`s only at this _z_-index (`drawOverLayers`) or under the layers (`drawUnderLayers`). However, you can easily achieve the desired result by subclassing `VerticalAxis` and drawing the `Decoration` from there, before the `VerticalAxis` labels are drawn. If the `Decoration` uses only `drawOverLayers`, which is the case for `HorizontalLine` and `HorizontalBox`, then this will look as follows:

```kt
override fun drawOverLayers(context: CartesianDrawingContext) {
    // Draw your `Decoration` via `Decoration.drawOverLayers`.
    super.drawOverLayers(context)
}
```

When using this approach, you should not add the `Decoration` to the `CartesianChart` via `decorations`, as this would result in duplication.

**Reply by my-waiwai** [reply]
Thanks for the reply. I'll give this a try when I have a chance to get back to working on the charts.
